gotham/src/handler/assets/mod.rs

```diff
@@ -518,33 +518,38 @@ mod tests {
         let path = "resources/test/assets/doc.html";
         let test_server =
             TestServer::new(build_simple_router(|route| route.get("/").to_file(path))).unwrap();
+        let test_client = test_server.client();
 
         let etag = File::open(path)
             .and_then(|file| file.metadata())
             .map(|meta| super::entity_tag(&meta).expect("entity tag"))
             .unwrap();
 
         // matching etag
-        let response = test_server
-            .client()
-            .get("http://localhost/")
-            .with_header(
+        let mut request = test_client.get("http://localhost/");
+
+        {
+            request.headers_mut().insert(
                 IF_NONE_MATCH,
                 HeaderValue::from_bytes(etag.as_bytes()).unwrap(),
-            ).perform()
-            .unwrap();
+            );
+        }
+
+        let response = test_client.perform(request).unwrap();
 
         assert_eq!(response.status(), StatusCode::NOT_MODIFIED);
 
         // not matching etag
-        let response = test_server
-            .client()
-            .get("http://localhost/")
-            .with_header(
+        let mut request = test_client.get("http://localhost/");
+
+        {
+            request.headers_mut().insert(
                 IF_NONE_MATCH,
                 HeaderValue::from_bytes("bogus".as_bytes()).unwrap(),
-            ).perform()
-            .unwrap();
+            );
+        }
+
+        let response = test_client.perform(request).unwrap();
```

Comment:
In this section (and ones like it) I'm not sure I see the advantage. I can see reusing the test_client, but headers_mut().insert seems like a step back from with_header. What am I missing?

Reply:
You're not actually missing anything - originally I removed `with_header`, so I had to rewrite this. 

Then I put it back and forgot to put this back, I'll do that now :p